src/StoreKeeper/WooCommerce/B2C/Tools/TaskRateCalculator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C\Tools;
+
+use StoreKeeper\WooCommerce\B2C\Models\TaskModel;
+
+class TaskRateCalculator
+{
+    private $tasks;
+    private $minuteMetrics;
+    private $startDateTime;
+    private $endDateTime;
+
+    public function __construct(array $tasks, $now = null, $minuteMetrics = 60)
+    {
+        $this->endDateTime = $now;
+        if (is_null($now)) {
+            $this->endDateTime = date('Y-m-d H:i:s');
+        }
+        $this->minuteMetrics = $minuteMetrics;
+        $this->startDateTime = date('Y-m-d H:i:s', strtotime("{$this->endDateTime} -{$minuteMetrics} minutes"));
+        $this->tasks = $tasks;
```

Comment:
You only use `$this->tasks` to count the total of them, quite overkill to load the all from db, just use mysql `count`

Reply:
Removed `$this->tasks` and added `countTasksByCreatedDateTimeRange`.